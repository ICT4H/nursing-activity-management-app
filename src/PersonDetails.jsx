import React, {Component} from 'react';

class PersonDetails extends Component {
  render() {
    let person = this.props.person;
    return (<div className={"PersonDetails"}>
      <p>
        {person.name}
      </p>
      <p>
        {person.gender}
      </p>
      <p>
        {person.age + " yrs"}
      </p>
    </div>)
  }
}

export default PersonDetails
