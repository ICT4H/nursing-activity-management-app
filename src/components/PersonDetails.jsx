import React, {Component} from 'react';

class PersonDetails extends Component {
  render() {
    let person = this.props.person;
    return (<div className={this.props.className}>
      <div>
        {person.name}
      </div>
      <div>
        {person.gender}
      </div>
      <div>
        {person.age + " yrs"}
      </div>
    </div>)
  }
}

export default PersonDetails
