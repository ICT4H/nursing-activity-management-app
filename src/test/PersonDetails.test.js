import React from 'react';
import PersonDetails from '../PersonDetails';
import renderer from 'react-test-renderer';

test('Given person object with name,gender and age gives all in "p" tag and wrapped around with "div" tag', () => {
  let person={name:"Cally Cardena", gender:"male",age:33};
  const component = renderer.create(
      <PersonDetails person={person} className={"personDetails"}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});