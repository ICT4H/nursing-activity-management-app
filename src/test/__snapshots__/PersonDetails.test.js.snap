// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given person object with name,gender and age gives all in "p" tag and wrapped around with "div" tag 1`] = `
<div
  className="PersonDetails"
>
  <p>
    Cally Cardena
  </p>
  <p>
    male
  </p>
  <p>
    33 yrs
  </p>
</div>
`;
